### ISSUE TITLE
[BUG] `vmap` when a circuit has `AmplitudeEmbedding` on a subset of qubits fails

### ISSUE THREAD

**Issue by ankit27kh** [PennyLaneAI/pennylane#5214]
### Expected behavior

If `AmplitudeEmbedding` is used on a subset of wires, the circuit should still work when more gates are applied to other qubits.

### Actual behavior

In this case, the execution fails if any other qubit in the circuit is used,

### Source code

```shell
dev = qml.device('default.qubit', wires=4, shots=None)


@qml.qnode(dev, interface='jax')
def circ(x):
    qml.AmplitudeEmbedding(x, wires=range(2), pad_with=0, normalize=True)
    qml.Hadamard(2)
    return qml.expval(qml.PauliX(2))


X = jnp.asarray(np.random.random([5, 4]))
print(circ(X[0]), 'This works')
print(jax.vmap(circ)(X), 'This does not work')
```

### Tracebacks

```shell
Traceback (most recent call last):
  File "/plugins/python/helpers/pydev/pydevconsole.py", line 364, in runcode
    coro = func()
  File "<input>", line 1, in <module>
  File "/plugins/python/helpers/pydev/_pydev_bundle/pydev_umd.py", line 197, in runfile
    pydev_imports.execfile(filename, global_vars, local_vars)  # execute the script
  File "/plugins/python/helpers/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "/t.py", line 18, in <module>
    print(jax.vmap(circ)(X), 'This does not work')
  File "/lib/python3.10/site-packages/pennylane/qnode.py", line 1039, in __call__
    res = qml.execute(
  File "/lib/python3.10/site-packages/pennylane/interfaces/execution.py", line 648, in execute
    results = inner_execute(tapes)
  File "/lib/python3.10/site-packages/pennylane/interfaces/execution.py", line 261, in inner_execute
    return cached_device_execution(tapes)
  File "/lib/python3.10/site-packages/pennylane/interfaces/execution.py", line 383, in wrapper
    res = list(fn(tuple(execution_tapes.values()), **kwargs))
  File "/lib/python3.10/site-packages/pennylane/devices/default_qubit.py", line 523, in execute
    results = tuple(
  File "/lib/python3.10/site-packages/pennylane/devices/default_qubit.py", line 524, in <genexpr>
    simulate(
  File "/lib/python3.10/site-packages/pennylane/devices/qubit/simulate.py", line 240, in simulate
    state, is_state_batched = get_final_state(circuit, debugger=debugger, interface=interface)
  File "/lib/python3.10/site-packages/pennylane/devices/qubit/simulate.py", line 123, in get_final_state
    state = create_initial_state(sorted(circuit.op_wires), prep, like=INTERFACE_TO_LIKE[interface])
  File "/lib/python3.10/site-packages/pennylane/devices/qubit/initialize_state.py", line 45, in create_initial_state
    return qml.math.asarray(prep_operation.state_vector(wire_order=list(wires)), like=like)
  File "/lib/python3.10/site-packages/pennylane/ops/qubit/state_preparation.py", line 230, in state_vector
    ket[indices] = op_vector
jax.errors.TracerArrayConversionError: The numpy.ndarray conversion method __array__() was called on traced array with shape float32[2,2].
See https://jax.readthedocs.io/en/latest/errors.html#jax.errors.TracerArrayConversionError
```

### System information

```shell
Name: PennyLane
Version: 0.34.0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/PennyLaneAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /home/keshav/tools/anaconda3/envs/credit_card/lib/python3.10/site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, requests, rustworkx, scipy, semantic-version, toml, typing-extensions
Required-by: PennyLane-Lightning
Platform info:           Linux-6.5.0-17-generic-x86_64-with-glibc2.35
Python version:          3.10.13
Numpy version:           1.26.3
Scipy version:           1.11.4
Installed devices:
- lightning.qubit (PennyLane-Lightning-0.34.0)
- default.gaussian (PennyLane-0.34.0)
- default.mixed (PennyLane-0.34.0)
- default.qubit (PennyLane-0.34.0)
- default.qubit.autograd (PennyLane-0.34.0)
- default.qubit.jax (PennyLane-0.34.0)
- default.qubit.legacy (PennyLane-0.34.0)
- default.qubit.tf (PennyLane-0.34.0)
- default.qubit.torch (PennyLane-0.34.0)
- default.qutrit (PennyLane-0.34.0)
- null.qubit (PennyLane-0.34.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by trbromley** [comment]
Thanks for posting @ankit27kh! Trying the code myself I don't get an error, I get:
```
0.9999999 This works
[0.9999999  0.99999994 0.9999999  1.0000001  0.99999994] This does not work
```
I am running with the latest version of JAX (0.4.24).

**Comment by ankit27kh** [comment]
Hey @trbromley, I have updated my JAX to version 0.4.24, but I am still getting this error.

**Comment by timmysilv** [comment]
Hi @ankit27kh, you are correct - this was not working in the latest release of PennyLane (v0.34). Fortunately, the fix has already been merged to master (#5028) and will be available in the next release, coming March 5.

Until then, you can install the latest `master` of PennyLane to have the fix included using `pip install git+https://github.com/PennyLaneAI/pennylane.git`, or [get a local editable version](https://docs.pennylane.ai/en/stable/development/guide/installation.html#installation) of PennyLane.

Let me know if this helped!

**Comment by trbromley** [comment]
@ankit27kh, let us know if you are still running into any issue here, but I'll close this issue for now.
